src/main/java/org/alliancegenome/curation_api/services/GeneService.java

```diff
@@ -123,6 +121,12 @@ public void removeOrDeprecateNonUpdated(Long id, String loadDescription) {
 					anyReferencingEntities = true;
 				}
 			}
+			List<Long> referencingGeneExpressionAnnotations = geneDAO.findReferencingGeneExpressionAnnotations(id);
+			if (referencingGeneExpressionAnnotations != null) {
```

Comment:
CollectionUtils.isNotEmpty() allows you to do the null check and size check in one go

Reply:
Fixed